subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationResultWriter.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import com.google.common.collect.ImmutableList;
+import org.gradle.api.UncheckedIOException;
+import org.gradle.api.file.RelativePath;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.gradle.internal.UncheckedException.unchecked;
+
+public class TransformationResultWriter {
```

Comment:
Given that the class can both read and write transform results, I think it's more like a serializer than a writer.

Reply:
I renamed it to `*Serializer`.